gpytorch/kernels/kernel.py

```diff
@@ -22,7 +22,14 @@ def __init__(self, postprocess_script=default_postprocess_script):
         self._postprocess = postprocess_script
 
     def _sq_dist(self, x1, x2, postprocess, x1_eq_x2=False):
-        if hasattr(torch, 'cdist') and x1.dim() < 3 and x1.size(-2) <= 512 and x2.size(-2) <= 512:
+        if (
+            hasattr(torch, 'cdist')
```

Comment:
Note that we can remove the `x1.dim() < 3` check since `torch.cdist` does support batch mode now. Or is there a reason to keep this @jacobrgardner, @gpleiss?

Also, interestingly gradients are fine for batched tensors. So we can use torch.cdist for small inputs even on the GPU if batched.

Finally how about defining the following helper so we can keep this logic in one place?

```
def _use_cdist(x1: Tensor, x2: Tensor) -> bool:
    return (
        hasattr(torch, 'cdist')
        and x1.size(-2) <= 512
        and x2.size(-2) <= 512
        # TODO: remove when https://github.com/pytorch/pytorch/issues/22353 is fixed
        and not (x1.is_cuda or x2.is_cuda)
    )
```

Reply:
I'm generally inclined to relegate cdist to the very small non batch setting because of the extreme performance issues with it on larger matrices (e.g., it is still up to 40x slower than our custom distance code).

I would be okay with a helper for when to use cdist. My hope is that the PyTorch team can eventually just make cdist the superior option in all settings.

Let me know when you want me to put 0.3.4 out.

Reply:
> My hope is that the PyTorch team can eventually just make cdist the superior option in all settings.

At the very least one could take the non-cdist code here as inspiration and port the logic into ATen (possibly with additional optimizations). 

> Let me know when you want me to put 0.3.4 out.

Let me do some digging how perf compares in batch mode, then I'll put up a PR with a helper that cleans this up. After that goes in we can put out 0.3.4.